packages/grafana-ui/src/components/Logs/LogRows.tsx

```diff
@@ -22,45 +22,34 @@ export interface Props extends Themeable {
   rowLimit?: number;
   onClickLabel?: (label: string, value: string) => void;
   getRowContext?: (row: LogRowModel, options?: any) => Promise<any>;
+  previewLimit?: number;
 }
 
 interface State {
-  deferLogs: boolean;
   renderAll: boolean;
 }
 
 class UnThemedLogRows extends PureComponent<Props, State> {
-  deferLogsTimer: number | null = null;
   renderAllTimer: number | null = null;
 
   state: State = {
-    deferLogs: true,
     renderAll: false,
   };
 
   componentDidMount() {
     // Staged rendering
-    if (this.state.deferLogs) {
-      const { data } = this.props;
-      const rowCount = data && data.rows ? data.rows.length : 0;
-      // Render all right away if not too far over the limit
-      const renderAll = rowCount <= PREVIEW_LIMIT * 2;
-      this.deferLogsTimer = window.setTimeout(() => this.setState({ deferLogs: false, renderAll }), rowCount);
-    }
-  }
-
-  componentDidUpdate(prevProps: Props, prevState: State) {
-    // Staged rendering
-    if (prevState.deferLogs && !this.state.deferLogs && !this.state.renderAll) {
+    const { data, previewLimit } = this.props;
+    const rowCount = data ? data.rows.length : 0;
+    // Render all right away if not too far over the limit
+    const renderAll = rowCount <= (previewLimit || PREVIEW_LIMIT) * 2;
```

Comment:
Nitpick: might want to extract that into separate function and used it everywhere?
```suggestion
    const renderAll = rowCount <= this.getPreviewLimit() * 2;
```